Providers/Modules/nxOMSAuditdPlugin/Scripts/OMSAuditdPlugin.sh

```diff
@@ -58,28 +76,59 @@ EOF
         service auditd start
     fi
 
-    # Give auoms a chance to start or stop
-    sleep 15
+    # Give auoms collector a chance to start or stop
+    sleep 5
 
-    if [ $1 = "yes" ]; then
-        # Make sure auoms started
-        pgrep -f $AUOMS_BIN 2>&1 >/dev/null
+    if [ "$1" = "running" -o $1 = "enabled" ]; then
+        # Make sure auoms collector started
+        pgrep -x -U 0 auoms 2>&1 >/dev/null
         if [ $? -ne 0 ]; then
             return 2
         fi
     else
-        # make sure auoms stopped
-        pgrep -f $AUOMS_BIN 2>&1 >/dev/null
+        # make sure auoms collector stopped
+        pgrep -x -U 0 auoms 2>&1 >/dev/null
         if [ $? -eq 0 ]; then
-            pkill -KILL -f $AUOMS_BIN
-            pgrep -f $AUOMS_BIN 2>&1 >/dev/null
+            pkill -KILL -x -U 0 auoms
+            pgrep -x -U 0 auoms 2>&1 >/dev/null
             if [ $? -eq 0 ]; then
                 return 2
             fi
         fi
     fi
 }
 
+get_auoms_state () {
+    $AUOMSCTL_BIN state 2>/dev/null
+    if [ $? -eq 5 ]; then
+        return 1
+    fi
+    return 0
+}
+
+set_auoms_state () {
+    if [ "$1" = "running" -o $1 = "enabled" ]; then
+        $AUOMSCTL_BIN enable
+    else
+        $AUOMSCTL_BIN disable
+    fi
+}
+
+get_audit_state () {
+    if [ -e $AUOMSCTL_BIN ]; then
+        $AUOMSCTL_BIN -s
+    else
+        /sbin/auditctl -s
+        RET=$?
+        if [ $RET -eq 139 ]; then
+            # On Debian 8, this command succeeds in outputing the results, but segfaults and returns 139
```

Comment:
>On Debian 8 [](start = 14, length = 11)

Is this condition hiding 139 on other OS platforms?

Reply:
This condition ensures that if auditctl terminates with a SIGSEGV it will not result in the overall failure of the script. The only place where this is happening is Debian 8.